index.js

```diff
@@ -288,7 +288,9 @@ function send (req, res, status, headers, message) {
     res.removeHeader('Content-Range')
 
     // response headers
-    setHeaders(res, headers)
+    for (const [key, value] of Object.entries(headers ?? {})) {
```

Comment:
We probably want to do some bench-marking on this. I don't want to armchair micro-benchmark on this, but I am pretty sure that this is making a unnecessary array per header and may be slower than the array from. Either way, these are the kind of changes that I want us to have a easy and reliable way to truly test the perf of before landing..